bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/ReplicationWorker.java

```diff
@@ -451,6 +455,10 @@ boolean isRunning() {
         return workerRunning && workerThread.isAlive();
     }
 
+    boolean isInReadOnlyMode() {
```

Comment:
Actually, in this case it's really about the ReplicationWorker to be aware that the target bookie has gone into read-only mode. The replicator needs to wait until the bookie comes back into rw mode